File: app/src/main/java/com/example/hbl/kotlin/coderead/HtmlParser.java
package com.example.hbl.kotlin.coderead;

import android.content.Context;
import android.support.v4.graphics.ColorUtils;

import com.example.hbl.kotlin.dir.FileNode;
import com.loopeer.codereader.R;

import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

import io.reactivex.Observable;

public class HtmlParser {
    public static String buildHtmlContent(Context context, String code, String filepatch
            , String fileName) {
        for (; ; ) {
            try {
                InputStream inputStream = context.getAssets().open("code.html");
                byte[] localObject = new byte[inputStream.available()];
                inputStream.read( localObject);
                inputStream.close();
                String localHtml = new String(localObject);
                jsFile = localHtml
                        .replace("!FONT_SIZE!"
                                , String.format("<style>.code .syntaxhighlighter { font-size: %.2fpx !important; }</style>"
                                        , new Object[]{Float.valueOf(PrefUtils.getPrefFontSize(context))}))
                        .replace("!FILENAME!"
                                , fileName)
                        .replace("!BRUSHJSFILE!", jsFile)
                        .replace("!SYNTAXHIGHLIGHTER!"
                                , localStringBuilder.toString())
                        .replace("!JS_FIX_HSCROLL!", temp);
                temp = "<link type='text/css' rel='stylesheet' href='style_menlo.css'/>";
                return jsFile
                        .replace("!STYLE_MENLO!", PrefUtils.getPrefMenlofont(context) ? temp : "")
                        .replace("!THEME!", PrefUtils.getPrefTheme(context))
                        .replace("!CODE!", paramString1)

                        .replace("!WINDOW_BACK_GROUND_COLOR!"
                                , ColorUtils.getColorString(context, R.color.code_read_background_color));
            } catch (IOException e) {
                throw new RuntimeException(e);
            } finally {

            }
        }
    }
    public static String getFileCode(final FileNode node){
        Observable.just(node)
                .map
        }

    }
}

File: app/src/main/java/com/example/hbl/kotlin/coderead/HtmlParser.kt
package com.example.hbl.kotlin.coderead

import android.text.TextUtils
import com.example.hbl.kotlin.andex.ext
import com.example.hbl.kotlin.dir.FileNode
import com.example.hbl.kotlin.extensions.log
import com.example.hbl.kotlin.extensions.select
import io.reactivex.Observable
import java.io.BufferedReader
import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.InputStreamReader
import java.util.*

/**
 * Created by hbl on 2017/5/26.
 */
class HtmlParser {
    companion object {
        fun getFileCode(node: FileNode): String? {
            Observable.just(node)
                    .filter { !isBlackFile(ext(node.name!!)) }
                    .map {
                        var fileInputStream:FileInputStream?=null
                        try {
                            fileInputStream = FileInputStream(it.absolutePath)
                        } catch (e: FileNotFoundException) {
                            log(e.message!!)
                        }
//                        var finalStream= select(fileInputStream==nul)
                        val sb = StringBuilder()
                        val bufferedReader = BufferedReader(InputStreamReader(, "UTF-8"))
                        while (true) {
                            bufferedReader.readLine()?.let { sb.append(it).append("\n") } ?: break
                        }
                        bufferedReader.close()
                        TextUtils.htmlEncode(sb.toString())
                    }
                    .map


        }

        val FILE_BLACKLIST = arrayOf("hprof", "apk", "jar", "so")

        fun isBlackFile(name: String?): Boolean {
            if (name != null) return Arrays.asList(*FILE_BLACKLIST).contains(name.toLowerCase())
            return false
        }
    }
}